pkg/generate/ack/hook.go

```diff
@@ -110,6 +112,11 @@ readOne call inside AWSResourceManager.LateInitialize() method
 The "late_initialize_post_read_one" hooks are called AFTER making the
 readOne call inside AWSResourceManager.LateInitialize() method
 
+The "is_synced_pre_check" hooks are called BEFORE the code that checks whether
+a resource is sync or not.
+
+The "is_synced_post_check" hooks are called AFTER the code that checks whether
+a resource is sync or not.
```

Comment:
Let's leave these hook points out for the time being. I'm not yet convinced we will need them.